### ISSUE TITLE
iPad landscape mode with open menu

### ISSUE THREAD

**Issue by lluisgerard** [erichoracek/MSDynamicsDrawerViewController#114]
I will modify the code to be able to show the menu on one side and the content on the other when you are on landscape mode (on iPad see Youtube app).
This is already done on other menu libraries like https://github.com/gotosleep/JASidePanels

![68747470733a2f2f696d672e736b697463682e636f6d2f32303132303332322d7474753935316e6662386370643574693572316e6938343238792e6a7067](https://cloud.githubusercontent.com/assets/1152968/3148612/85fea210-ea62-11e3-8cc2-b6bd4d55c714.jpeg)

I think is not already possible and this would be a nice feature to have. I'm on a schedule so I can't dedicate too much time but will try to do it, will post any updates here.

**Comment by lluisgerard** [comment]
Actually, I have a working version with the behavior but have some issues with the firstload and maintaining the pane open and accepting touches. In the end it doesn't seems that hard to achieve playing with `willAnimateRotationToInterfaceOrientation` method :)

**Comment by lluisgerard** [comment]
Actually I rewrote it all in Swift... just kidding :)

**Comment by lluisgerard** [comment]
I have a working version here, and it works perfectly fine with the project I'm working on:
https://github.com/lluisgerard/MSDynamicsDrawerViewController

I cannot share that project/screenshots so it would be nice to create a new example targeted to iPad to show this feature.

**IMPORTANT**:

```
CGFloat const MSDynamicsDrawerDefaultOpenStateRevealWidthHorizontal
```

I changed this constant from 267.0 to 256.0 to make it fit the width of the contentView (menuRevealWidth + contentWidth) on landscape. It would be better to make it dynamic so it adapts to whatever width you want to reveal (though is easier to make the width on landscape mode fit exactly the fit on portrait width).

I expect to improve it a little bit (and contributions are welcome). Maybe it would be better to push it all on a new branch and upload it to the main repo, if there is some interest on it.

**Comment by lluisgerard** [comment]
If you are using CocoaPods and want to test it, you can do so changing this line:

```
pod 'MSDynamicsDrawerViewController'
```

to:

```
pod 'MSDynamicsDrawerViewController', :git => 'https://github.com/lluisgerard/MSDynamicsDrawerViewController.git'
```

And run `pod install` to download it from the other repo.

**Comment by nfranke** [comment]
If you start the iPad in landscape mode, it goes crazy: the menu is almost full width and then clicking on the open draw icon removes the pane view entirely. This patch also breaks the right menu (both the button and swipe to drag.) Perhaps that's what you said in the second message, but it wasn't clear if you thought that was fixed.

I'd also suggest creating a new read property to reduce repetition:

```
- (BOOL)leftDirectionShouldStayOpen
{
    return self.leftDirectionShouldStayOpenInLandscape && (UIInterfaceOrientationIsLandscape(self.paneViewController.interfaceOrientation));
}
```

**Comment by lluisgerard** [comment]
@nfranke exactly as you say, I haven't tested it with a right panel and I want to make it cleaner (thanks for the suggestion! :+1: ).

What I know for sure is that it works perfectly in my project. Actually, we will be releasing the app soon (is well tested with different iPads and people) and it can start landscape or portrait. I'm with this release right now and can't clean the code these days.

**Comment by nfranke** [comment]
It seems to greatly depend on where one sets leftDirectionShouldStayOpenInLandscape to YES. I initially tried doing so right after configuring the dynamicsDrawerViewController, but that make it go crazy. Moving it to after my initial call to setPaneViewController improved things. I had to add calls to setPaneState in setLeftDirectionShouldStayOpenInLandscape to get it working reliably. I see you had code in viewDidLoad to do this, but it was never called for me since viewDidLoad is called before I set leftDirectionShouldStayOpenInLandscape since I load from a storyboard.

If you don't want to change MSDynamicsDrawerDefaultOpenStateRevealWidthHorizontal, then you can add this method and call that from setLeftDirectionShouldStayOpenInLandscape and willAnimateRotationToInterfaceOrientation:

```
- (void)setPaneViewFrame {
    UIInterfaceOrientation ui_orientation = [[UIApplication sharedApplication] statusBarOrientation];
    CGRect frame = [UIScreen mainScreen].bounds;
    if (UIInterfaceOrientationIsLandscape(ui_orientation)) {
        CGFloat t = frame.size.width;
        frame.size.width = frame.size.height;
        frame.size.height = t;
    }

    frame.origin.x = [self openStateRevealWidth];
    frame.origin.y = 0;
    frame.size.width -= [self openStateRevealWidth];
    self.paneView.frame = frame;
}
```

Removing the "return NO" lines from the gesture recognizer-related code allows me to slide from one side to the other, but the right pane doesn't appear. Tapping the right reveal button makes it go crazy.

**Comment by lluisgerard** [comment]
@nfranke oh, sure, I haven't tested it on storyboards/xibs... I only have code in my project :dancer:

**Comment by nfranke** [comment]
I wonder if wrapping the MSDynamicsDrawerViewController with left menu inside another MSDynamicsDrawerViewController with a right menu would work? I'll have to try that.

**Comment by will-tm** [comment]
Does someone has any news on that subject ?

**Comment by pnull** [comment]
@lluisgerard @nfranke any news on that - i really like the project and would absolutely LOVE to have it work with an always open drawer in landscape :heart:

**Comment by lluisgerard** [comment]
Hi there. 
I ended up using this to build this app: http://www.northkiteboarding.com and so far I think it was working when I handed the code. The problem is that I stepped out at the project weeks before the release and maybe they changed something.

The code I put here worked: https://github.com/lluisgerard/MSDynamicsDrawerViewController but it's outdated and may have some bugs. One thing that could be done is a merge from @erichoracek code and take care of `leftDirectionShouldStayOpenInLandscape`.

Right now I'm working on other apps entirely in Swift, wish I had some time to fix this guys but I can't see it happen on a near future.
